doc/ecosystem.md

```diff
@@ -0,0 +1,131 @@
+language. It is already well on its way toward this goal.
+
+#### Xarray
+
+[Xarray][] is an open source project and Python package that introduces labels in the
+form of dimensions, coordinates, and attributes on top of raw NumPy-like arrays, which
+allows for more intuitive, more concise, and less error-prone user experience.
+
+### Optional dependencies
+
+#### IPython
+
+[IPython][] provides a rich toolkit to help you make the most of using Python
+interactively. Its main components are a powerful interactive Python shell and a Jupyter
+kernel to work with Python code in Jupyter notebooks and other interactive frontends.
+
+PyGMT relies on IPython to provide a rich interactive experience in Jupyter notebooks.
+
+#### GeoPandas
+
+[GeoPandas][] is an open source project to make working with geospatial data in Python
+easier. GeoPandas extends the datatypes used by [pandas][] to allow spatial operations
+on geometric types. Geometric operations are performed by [Shapely][]. GeoPandas further
+depends on [pyogrio][] for file access and [Matplotlib][] for plotting.
+
+PyGMT doesn't directly rely on GeoPandas, but provides support of GeoPandas's data
+structure, {class}`geopandas.GeoDataFrame` and {class}`geopandas.GeoSeries`, which can
+be directly used in data processing and plotting functions/methods of PyGMT.
+
+#### contextily
+
+[contextily][] is a small Python package to retrieve tile maps from the internet. It can
+add those tiles as basemap to matplotlib figures or write tile maps to disk into
+geospatial raster files. Bounding boxes can be passed in both WGS84 (EPSG:4326) and
+Spheric Mercator (EPSG:3857).
+
+In PyGMT, {func}`pygmt.datasets.load_tile_map` and {class}`pygmt.Figure.tilemap` rely
+on it.
+
+#### rioxarray
+
+[rioxarray][] is a geospatial [Xarray][] extension powered by rasterio. Built on top of
+[Rasterio][], it enables seamless reading, writing, and manipulation of multi-dimensional
+arrays with geospatial attributes such as coordinate reference systems (CRS) and spatial extent
+(bounds).
+
+Currently, PyGMT relies on [rioxarray][] to save multi-band rasters to temporary files
+in GeoTIFF format, to support processing and plotting 3-D {class}`xarray.DataArray`
+images.
+
+```{note}
+We're working towards removing the dependency of the [rioxarray][] package in
+[PR #3468](https://github.com/GenericMappingTools/pygmt/pull/3468).
+```
+
+#### PyArrow
+
+[Apache Arrow][] is a development platform for in-memory analytics. It contains a set of
+technologies that enable big data systems to process and move data fast. It specifies a
+standardized language-independent columnar memory format for flat and hierarchical data,
+organized for efficient analytic operations on modern hardware. The Arrow Python bindings
+(also named "PyArrow") have first-class integration with NumPy, pandas, and built-in
+Python objects. They are based on the C++ implementation of Arrow.
+
+```{note}
```

Comment:
This note is copied from https://www.pygmt.org/dev/install.html with updates from PR #2933. 

The plan is to remove the note from https://www.pygmt.org/dev/install.html. See PR #3506.

Reply:
It's likely this PR will be merged before PR #2933, so I've reverted the changes in PR #2933 (770968e96593735890532fc3de09931b5b78ecef) and use the original notes from the installation guides.